### PR TITLE
Webscan: Migration + Clean Up

### PR DESCRIPTION
**Pull request by payoub125**
# Done

- Removed Tools that got transferred to `Methodwebtest`
    - Header Misconfiguration Modules (ie.Cors)
    - Web Server Specific Modules
    - Path Transversal (Fuzz) Tool

- Fernified `webserver probe`
  
- Updated Command Structure
  - `webscan fingerprint` -> `webscan url fingerprint `
  - `webscan fuzz ratelimit` -> `webscan webserver ratelimit`

- Update Repo Structure
  - All files under internal are now are nested in a manor that mirrors the CLI Command structure
  - All fern files are now are nested in a manor that mirrors the CLI Command structure so that we dont have naming conflicts or need to generate 'strange' names
  - No command uses the `utils.request` function directly

- Cleaned up `App Fingerprint` code + fern
  - Modules use `utils.request` function
  - New fern sends back consistent data across modules 

- Clean up Headergrab
  - Now  uses `utils.request` function

- Updated Documentation

### CODE REVIEW

**Comment by methodsecuritysj** [comment]
@payoub125 ready to review?

**Comment by payoub125** [comment]
> @payoub125 ready to review?

Yes @methodsecuritysj thank you. Let me know if I need to document/explain whats happening in a better manner

File: cmd/root.go [hunk 1]
```diff
@@ -20,17 +20,18 @@ import (
...
-	ProbeCmd     *cobra.Command
-	SpiderCmd    *cobra.Command
-	VulnCmd      *cobra.Command
-	AppCmd       *cobra.Command
+	Version         string
+	RootFlags       config.RootFlags
+	OutputConfig    writer.OutputConfig
+	OutputSignal    signal.Signal
+	RootCmd         *cobra.Command
+	VersionCmd      *cobra.Command
```
Comment (sean-hacker) [inline cmd/root.go]: Not a blocker but we don't actually need to store each command on the struct. We can just attach them to the root command and only store the root.

File: docs/docs/pagecapture.md
Comment (sean-hacker) [inline docs/docs/pagecapture.md:1]: I do think we need a bit more verbose documentation here that outlines what these various commands actually do and how they differ from one another, but I don't view it as blocking.

**Review by sean-hacker — DISMISSED** [review]

File: docs/docs/pagecapture.md
Comment (payoub125) [reply docs/docs/pagecapture.md:1]: ack

File: cmd/root.go (hunk 1 above)
Comment (payoub125) [reply cmd/root.go]: ack

**Review by steven-rand — APPROVED** [review]

**Pull request merged**